application_form/api/views.py

```diff
@@ -11,3 +16,23 @@ class ApplicationViewSet(AuditLoggingModelViewSet):
     permission_classes = [permissions.IsAuthenticated]
     lookup_field = "external_uuid"
     http_method_names = ["post"]
+
+
+class ListProjectReservations(GenericAPIView):
+    """
+    Public: Returns a list of the user's apartment reservations from a specific project.
+    """
+
+    serializer_class = ApartmentReservationSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    http_method_names = ["get"]
+
+    def get(self, request, project_uuid):
+        profile_uuid = request.user.profile.id
+        reservations = ApartmentReservation.objects.filter(
+            apartment__project__uuid=project_uuid,
+            application_apartment__application__profile__id=profile_uuid,
+        )
+        serializer = ApartmentReservationSerializer(data=reservations, many=True)
+        serializer.is_valid()
```

Comment:
This should now be needed as we are not handling user input data here but instead a model instance. DRF does not run `.is_valid()` when it returns from a `list` or `retrieve` method either.

Reply:
Good to know 👍  This is changed.